### PR TITLE
Feature/NEX-1529/remove test preview dependency

### PR DESCRIPTION
**Pull request by bziondik**
Related to: https://oat-sa.atlassian.net/browse/NEX-1529
Related to:
https://github.com/oat-sa/extension-tao-test/pull/348
https://github.com/oat-sa/extension-tao-testqti-previewer/pull/88

**Issue:**
In working branches 
"oat-sa/extension-tao-test": "dev-feature/NEX-1043/test-preview-base-branch",  
"oat-sa/extension-tao-testqti": "dev-feature/NEX-1043/test-preview-base-branch", 
an issue with building bundles because of new dependency taoQtiTestPreview

**Solution:**
Dynamically load taoQtiTestPreviewer adapter

**How to test:**

1. checkout brances "oat-sa/extension-tao-test": "dev-feature/NEX-1043/test-preview-base-branch",  "oat-sa/extension-tao-testqti": "dev-feature/NEX-1043/test-preview-base-branch", 
2. in taoQtiTestPreviwer checkout to develop
3. open Tests page
4. press Preview button
5. in taoQtiTestPreviwer checkout to feature/NEX-1043/test-preview-base-branch
6. reload Tests page
7. press Preview button

### CODE REVIEW

File: views/js/controller/creator/creator.js [hunk 1]
```diff
@@ -60,11 +64,29 @@ define([
...
+    const logger = loggerFactory('taoQtiTest/controller/creator');
+    let previewerFactory = null;
 
+    providerLoaderFactory()
+        .addList({
+            previwers: {
+                id: 'qtiTests',
+                module: 'taoQtiTestPreviewer/previewer/adapter/test/qtiTest',
+                bundle: 'taoQtiTestPreviewer/loader/qtiPreviewer.min',
+                category: 'previwers'
```
Comment (jsconan) [inline views/js/controller/creator/creator.js:80]: typo in names

```suggestion
            previewers: {
                id: 'qtiTests',
                module: 'taoQtiTestPreviewer/previewer/adapter/test/qtiTest',
                bundle: 'taoQtiTestPreviewer/loader/qtiPreviewer.min',
                category: 'previewers'
```

File: views/js/controller/creator/creator.js [hunk 2]
```diff
@@ -226,10 +248,14 @@ define([
...
-                                readOnly: false,
-                                fullPage: true
-                            });
+                            if (previewerFactory) {
+                                return previewerFactory.init(decodeURIComponent(testUri), {
+                                    readOnly: false,
+                                    fullPage: true
+                                });
+                            } else {
+                                feedback().error('Test Preview is not installed, please contact to your administrator.');
```
Comment (jsconan) [inline views/js/controller/creator/creator.js:257]:
```suggestion
                                feedback().error('Test Preview is not installed, please contact your administrator.');
```

**Review by jsconan — COMMENTED** [review]
After a quick glance, it looks good. But it would have been better to not hardcode the previewers.
Can't you rely on the backend registry instead?

There are a few typos to fix.

File: views/js/controller/creator/creator.js [hunk 3]
```diff
@@ -226,9 +229,17 @@ define([
                         if(isTestContainsItems()) {
                             const saveUrl = options.routes.save;
                             const testUri = saveUrl.slice(saveUrl.indexOf('uri=') + 4);
-                            return previewerFactory.init(decodeURIComponent(testUri), {
-                                readOnly: false,
-                                fullPage: true
+                            return previewerFactory(
+                                'qtiTest',
```
Comment (jsconan) [inline views/js/controller/creator/creator.js:233]: Is it not possible to have this name managed trough a config instead?

**Review by jsconan — COMMENTED** [review]
Looks good, but I would recommend not relying on hardcoded names. It is better to be able to configure it and rely on the platform to get the right provided to load

File: views/js/controller/creator/creator.js (hunk 3 above)
Comment (bziondik) [reply views/js/controller/creator/creator.js:233]: I am not sure that this should be in configs. 
https://github.com/oat-sa/extension-tao-item/blob/1b3fdb935f0298fc7dfd66d0d023c319199b7bfa/views/js/controller/items/action.js#L29 here also hardcoded
At what configs it should be put?

**Comment by bziondik** [comment]
@jsconan The configuration will be handled in a separate ticket

**Comment by habemuscode** [comment]
Manifest version?

**Review by habemuscode — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`

**Review by jsconan — APPROVED** [review]
Looks good so far.
Will need a rework to properly handle the previewer instead of hardcoding the name.

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**